### PR TITLE
VMware NSX-T SDN Install and Config updates

### PR DESCRIPTION
**Pull request by mburke5678**
Follow-up to https://github.com/openshift/openshift-docs/pull/14649

### CODE REVIEW

**Comment by mburke5678** [comment]
@aheslin FYI I created a PR for the Vmware nsxt issue. I thought it quicker to add my edits here. This is a first pass. LMK if we are going forward with this.

**Comment by vikram-redhat** [comment]
@mburke5678  in case you don't get a response from Ali, yes, we are going ahead with it. 

Does this supercede #14649?

File: install_config/configuring_nsxtsdn.adoc [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+master02.example.com ansible_ssh_host=192.168.220.3 OpenShift_node_group_name='node-config-master' OpenShift_ip=192.168.220.3
+master03.example.com ansible_ssh_host=192.168.220.4 OpenShift_node_group_name='node-config-master' OpenShift_ip=192.168.220.4
+node01.example.com ansible_ssh_host=192.168.220.5 OpenShift_node_group_name='node-config-infra' OpenShift_ip=192.168.220.5
+#node02.example.com ansible_ssh_host=192.168.220.6 OpenShift_node_group_name='node-config-infra' OpenShift_ip=192.168.220.6
+node03.example.com ansible_ssh_host=192.168.220.7 OpenShift_node_group_name='node-config-compute' OpenShift_ip=192.168.220.7
+node04.example.com ansible_ssh_host=192.168.220.8 OpenShift_node_group_name='node-config-compute' OpenShift_ip=192.168.220.8
+----
+
+.Procedure
+
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @vikram-redhat Do we want to refer people out to the OpenShift install prerequisite docs here and remove the command?

File: install_config/configuring_nsxtsdn.adoc [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+. Get the image name and retag it:
++
+----
+$ docker images
+$ docker image tag registry.local/xxxxx/nsx-ncp-rhel nsx-ncp
+----
+
+. Deploy the {product-title} cluster:
+----
+$ ansible-playbook -i hosts openshift-ansible/playbooks/deploy_cluster.yml
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @vikram-redhat Do we want to refer people out to the OpenShift install docs here and remove the command?

File: install_config/configuring_nsxtsdn.adoc [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+node03.example.com ansible_ssh_host=192.168.220.7 OpenShift_node_group_name='node-config-compute' OpenShift_ip=192.168.220.7
+node04.example.com ansible_ssh_host=192.168.220.8 OpenShift_node_group_name='node-config-compute' OpenShift_ip=192.168.220.8
+----
+
+.Procedure
+
+. Run the following command to install required software packages, if any, for {product-title}:
++
+----
+$ ansible-playbook -i hosts openshift-ansible/playbooks/prerequisites.yml
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @vikram-redhat Do we want to refer people out to the OpenShift install prerequisite docs here and remove the command?

**Comment by mburke5678** [comment]
@RobbieJVMW Can you review my changes? I want to make sure my efforts to change the wording and such to better fit the OpenShift docs doesn't change any technical meaning.
Also, I have a few questions in https://github.com/openshift/openshift-docs/pull/14649

**Comment by mburke5678** [comment]
@ousleyp Can you review this partner doc? They want it live by tomorrow for Summit.

File: architecture/topics/nsxt.adoc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+The VMware NSX-T Data Center provides a policy-based overlay network reproducing the complete set of Layer 2
+through Layer 7 networking services (such as switching, routing, access control, fire-walling, and QoS) in software for
+native {product-title} networking capabilities.
+
+The NSX-T components can be installed and configured as part of the Ansible installation procedure which integrates an {product-title} SDN
```
Comment (ahardin-rh) [inline architecture/topics/nsxt.adoc]: , which

File: architecture/topics/nsxt.adoc [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+into a datacenter-wide NSX-T virtualised network connecting bare metal, virtual machines, and {product-title} pods.
+See the xref:../../install_config/configuring_nsxtsdn.adoc#install-config-configuring-nsxt-sdn[Installation] section for information on how to install and deploy OpenShift Container Platform with VMware NSX-T.
```
Comment (ahardin-rh) [inline architecture/topics/nsxt.adoc]: s/OpenShift Container Platform/{product-title}

File: architecture/topics/nsxt.adoc [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+The *NSX-T Container Plug-In* (NCP) integrates {product-title} into an NSX-T Data Center, which is typically configured for the entire datacenter.
```
Comment (ahardin-rh) [inline architecture/topics/nsxt.adoc]: s/datacenter/data center

File: architecture/topics/nsxt.adoc [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+native {product-title} networking capabilities.
+
+The NSX-T components can be installed and configured as part of the Ansible installation procedure which integrates an {product-title} SDN
+into a datacenter-wide NSX-T virtualised network connecting bare metal, virtual machines, and {product-title} pods.
+See the xref:../../install_config/configuring_nsxtsdn.adoc#install-config-configuring-nsxt-sdn[Installation] section for information on how to install and deploy OpenShift Container Platform with VMware NSX-T.
+
+The *NSX-T Container Plug-In* (NCP) integrates {product-title} into an NSX-T Data Center, which is typically configured for the entire datacenter. 
+
+For information on the NSX-T Data Cetner architecture and administration, see the link:https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.4/installation/GUID-10B1A61D-4DF2-481E-A93E-C694726393F9.html[VMware NSX-T Data Center v2.4 documentation] and the link:https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.4/com.vmware.nsxt.ncp_openshift.doc/GUID-1D75FE92-051C-4E30-8903-AF832E854AA7.html[NSX-T NCP configuration] guides.
```
Comment (ahardin-rh) [inline architecture/topics/nsxt.adoc]: s/Cetner/Center 

Not sure Data Center is a proper noun here. If not, I think we can lower case it.

File: install_config/configuring_nsxtsdn.adoc [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+organizations to have complete visibility using a single SDN across the entire environment.
+
+For more information on how NSX-T integrates with {product-title}, see the xref:../architecture/networking/network_plugins.adoc#nsx-sdn[NSX-T SDN in Available SDN plug-ins].
+
+[[nsx-t-sdn-operations-workflow]]
+== Example Topology
+
+One typical use case is to have a Tier-0 (T0) router that connects the physical system with the virtual environment and a Tier-1 (T1) router to act as a default gateway for the {product-title} VMs.
+
+Each VM has two vNICs: One vNIC connects to the Management Logical Switch for accessing the VMs. The other vNIC connects to a Dump Logical Switch and is used by `nsx-node-agent` to uplink the Pod networking. For further details, refer to the link:https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.4/nsxt_24_ncp_openshift.pdf[NSX Container Plug-in for OpenShift].
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/refer to the/refer to

File: install_config/configuring_nsxtsdn.adoc [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+
+The LoadBalancer used for configuring {product-title} Routes and all project T1 routers and Logical Switches are created automatically during the {product-title} installation.
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/LoadBalancer/load balancer

File: install_config/configuring_nsxtsdn.adoc [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+[[nsx-t-sdn-operations-workflow]]
+== Example Topology
+
+One typical use case is to have a Tier-0 (T0) router that connects the physical system with the virtual environment and a Tier-1 (T1) router to act as a default gateway for the {product-title} VMs.
+
+Each VM has two vNICs: One vNIC connects to the Management Logical Switch for accessing the VMs. The other vNIC connects to a Dump Logical Switch and is used by `nsx-node-agent` to uplink the Pod networking. For further details, refer to the link:https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.4/nsxt_24_ncp_openshift.pdf[NSX Container Plug-in for OpenShift].
+
+The LoadBalancer used for configuring {product-title} Routes and all project T1 routers and Logical Switches are created automatically during the {product-title} installation. 
+
+In this topology the default {product-title} HAProxy Router is used for all infrastructure components such as Grafana, Prometheus, Console, Service Catalog, and others.
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: topology, the

File: install_config/configuring_nsxtsdn.adoc [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+Ensure that the DNS records for the infrastructure components point the infrastructure node IP addresses, because the HAProxy uses the host network namespace.
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: point to the infrastructure node IP addresses?

File: install_config/configuring_nsxtsdn.adoc [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+This works for infrastructure routes, but in order to avoid exposing the infrastructure nodes management IPs to the outside world, deploy application-specific routes to the NSX-T LoadBalancer.
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/LoadBalancer/load balancer

File: install_config/configuring_nsxtsdn.adoc [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+
+This example topology assumes you are using 3 {product-title} master virtual machines and 4 {product-title} worker virtual machines (two for infrastructure and two for compute).
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/3/three
s/4/four

File: install_config/configuring_nsxtsdn.adoc [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+
+* ESXi hosts requirements:
+** ESXi servers that host {product-title} node VMs must be NSX-T Transport Nodes.
++
+.NSX UI dislaying the Transport Nodes for a typical high availability environment: 
++
+image::nsxt-transportnodes.png[NSX Transport Nodes]
+
+* DNS requirements:
+** You must add a new entry to your DNS server with a wildcard to the infrastructure nodes. This allows load balancing by NSX-T or other third-party LoadBalancer. In the `hosts` file below, the entry is defined by the `openshift_master_default_subdomain` variable.
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/LoadBalancer/load balancer

File: install_config/configuring_nsxtsdn.adoc [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+** You must update your DNS server with the `openshift_master_cluster_hostname` & `openshift_master_cluster_public_hostname` variables.
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/&/and

File: install_config/configuring_nsxtsdn.adoc [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+
+* Virtual Machine requirements:
+** The {product-title} node VMs must have two vNICs:
+** A Management vNIC must be connected to the Logical Switch that is uplinked to the management T1 router.
+** The second vNIC on all VMs must be tagged in NSX-T so that the NSX Container Plugin (NCP) knows which port needs to be used as a parent VIF for all Pods running in a particular {product-title} node. The tags must be the following:
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/Plugin/Plug-in

File: install_config/configuring_nsxtsdn.adoc [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
++
+The following prerequisites need to be met in NSX:
++
+** A Tier-0 Router.
+** An Overlay Transport Zone.
+** An IP Block for POD networking.
+** Optionally, an IP Block for routed (NoNAT) POD networking.
+** An IP Pool for SNAT. By default the subnet given per Project from the Pod networking IP Block is routable only inside NSX-T. NCP uses this IP Pool to provide connectivity to the outside.
+** Optionally, the Top and Bottom firewall sections in a dFW (Distributed Firewall). NCP places the Kubernetes Network Policy rules between those two sections.
+** The Open vSwitch and CNI plugin RPMs need to be hosted on a HTTP server reachable from the {product-title} Node VMs (`http://websrv.example.com` in this example). Those files are included in NCP Tar file, which you can download from vmWare at link:https://my.vmware.com/web/vmware/details?downloadGroup=NSX-T-PKS-240&productId=673[Download NSX Container Plugin 2.4.0
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/plugin/plug-in
s/in NCP Tar file/in the NCP tar file
s/vmWare/VMware

File: install_config/configuring_nsxtsdn.adoc [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
+].
+
+* In the {product-title} Ansible hosts file, specify the following parameters to setup NSX-T as the network plug-in:
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/setup/set up

File: install_config/configuring_nsxtsdn.adoc [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
+docker image tag registry.local/2.4.0.12511604/nsx-ncp-rhel nsx-ncp
+----
+
+. Deploy the {product-title} cluster:
++
+----
+$ ansible-playbook -i hosts openshift-ansible/playbooks/deploy_cluster.yml
+----
++
+For more information on the {product-title} installation, see xref:../install/running_install.adoc#install-running-installation-playbooks[Installing OpenShift Container Platform].
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: s/OpenShift Container Platform/{product-title}

File: install_config/configuring_nsxtsdn.adoc [hunk 20]
```diff
@@ -0,0 +1,239 @@
...
+== Check NSX-T after {product-title} deployment
+
+After installing {product-title} and verifying the NCP and `nsx-node-agent-*` Pods:
+
+* Check the routing. Ensure that the Tier-1 routers were created during the installation and are linked to the Tier-0 router:
++
+.NSX UI dislaying showing the T1 routers
+image::nsxt-routing.png[NSX routing]
+
+* Observe the network traceflow and visibility. For example check the connection between 'console' and 'grafana'.
```
Comment (ahardin-rh) [inline install_config/configuring_nsxtsdn.adoc]: For example, check

**Comment by ahardin-rh** [comment]
@mburke5678 Just some minor comments from me. Otherwise, looks good!

File: architecture/topics/nsxt.adoc (hunk 6 above)
Comment (RobbieJVMW) [reply architecture/topics/nsxt.adoc]: "NSX-T Data Center" is the product name. :)

File: architecture/topics/nsxt.adoc (hunk 7 above)
Comment (RobbieJVMW) [reply architecture/topics/nsxt.adoc]: Agree in this context it should probably say "NSX-T Manager" rather than "NSX-T Data Center"

Comment (mburke5678) [reply architecture/topics/nsxt.adoc]: I believe that is the name of the product. 
https://www.vmware.com/products/nsx.html

File: install_config/configuring_nsxtsdn.adoc (hunk 9 above)
Comment (mburke5678) [reply install_config/configuring_nsxtsdn.adoc]: @ahardin-rh Is this a kubernetes object, here? They use LoadBalancer.

File: architecture/topics/nsxt.adoc (hunk 7 above)
Comment (RobbieJVMW) [reply architecture/topics/nsxt.adoc]: The *NSX-T Container Plug-In* (NCP) integrates {product-title} into an NSX-T Manager, which is typically configured for the entire datacenter.

File: install_config/configuring_nsxtsdn.adoc (hunk 17 above)
Comment (mburke5678) [reply install_config/configuring_nsxtsdn.adoc]: Did they change vmWare? Could sworn it was that way, but you are right....

File: install_config/configuring_nsxtsdn.adoc (hunk 19 above)
Comment (mburke5678) [reply install_config/configuring_nsxtsdn.adoc]: @ahardin-rh Do we want to change that one to {product-title}? It is the name of the section in the link.

**Comment by mburke5678** [comment]
@ahardin-rh Thanks for the great (and quick) review!

**Comment by mburke5678** [comment]
Per Robbie in email"
One of my NSX specialists have already taken a look at the preview site and didn’t flag anything.  He knows NSX far better than I so I would consider that a green light.

**Pull request merged**